aviary/mission/gasp_based/ode/time_integration_base_classes.py

```diff
@@ -7,23 +7,53 @@
 
 from aviary.mission.gasp_based.ode.params import ParamPort
 from aviary.variable_info.enums import Verbosity
+from aviary.variable_info.variable_meta_data import _MetaData
 
 
 def add_SGM_required_inputs(group: om.Group, inputs_to_add: dict):
-    blank_execcomp = om.ExecComp()
```

Comment:
What is the ultimate purpose of creating a "sink" component for these vars?

Reply:
the simupy integrator expects that all states be an input to the ODE and all state rates be an output from the ODE. If a state is not directly used in the EOM calculations, this lets you add it without changing the EOM.